astropy/coordinates/tests/test_spectral_coordinate.py

```diff
@@ -54,13 +55,16 @@ def assert_frame_allclose(frame1, frame2,
         assert_quantity_allclose(d1.norm(d1), d1.norm(d2), rtol=vel_rtol, atol=vel_atol)
 
 
-def get_greenwich_earthlocation():
-    """
-    A helper function to get an EarthLocation for greenwich (without trying to
-    do a download)
-    """
-    site_registry = EarthLocation._get_site_registry(force_builtin=True)
-    return site_registry.get('greenwich')
+@pytest.fixture(scope="module")
+def greenwich_earthlocation():
+    # An attempt might be made to access remote data.
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            "ignore",
+            category=AstropyUserWarning,
+            message="Could not access the online site list.",
+        )
+        return EarthLocation.of_site("Greenwich")
```

Comment:
How do we know this won't silently try to grab remote data?

Reply:
The relevant function for this question is https://github.com/astropy/astropy/blob/f4615bfafafa4bf75dd4a6b52de1695c1b9f8ea2/astropy/coordinates/earth.py#L508

By reading the function we can see that if the site registry has been accessed by a previous test then the data is simply looked up from the version of the registry that is already loaded in memory.

If the registry has not been initialized then the cache directory is searched. If the data is there then no Internet connection is attempted.

If there is no cached file then an attempt is made to fetch the data from the Internet. In remote data tests this is what we want, and the registry that will be loaded in memory will be the online version. In non-remote data tests the lookup fails because `astropy` has enabled the `remote_data_strict` option of `pytest-remotedata`. A warning is emitted and the small built-in registry is used instead. All we have to do is catch the warning in the fixture.

Greenwich observatory is listed both in the online registry and the built-in one, so we get the required data either way.

Reply:
Test that is not marked as remote data should not silently try to access the internet under certain condition. We do not want that ambiguity. Is there a way to make sure that this definitely will not try to access the internet? We previously force a `force_builtin=True` but you have removed that in this patch.

Reply:
There must be a better way to do this... Maybe it is time to figure out a solution for #11196 . Should not be this hard to switch back and forth.